packages/adapter/addon/rest.js

```diff
@@ -1301,9 +1301,11 @@ function fetchSuccessHandler(adapter, payload, response, requestData) {
 
 function fetchErrorHandler(adapter, payload, response, errorThrown, requestData) {
   let responseData = fetchResponseData(response);
+  payload = adapter.parseErrorResponse(payload);
+
   if (responseData.status === 200 && payload instanceof Error) {
```

Comment:
It is interesting the error you throw in your test is 422, but the check here is 200.  I thought there was a reason for this but I can't remember why.  Do you happen to know?

Reply:
After re-reding the code, it seems that there could be some case where the HTTP status is 200, but an error is thrown (for example due to a runtime error). In that case, we fall into the status=200 branch, so we don't need to parse payload there.